pkg/cri/opts/spec_linux.go

```diff
@@ -636,6 +636,17 @@ func WithPodNamespaces(config *runtime.LinuxContainerSecurityContext, sandboxPid
 	if namespaces.GetPid() != runtime.NamespaceMode_CONTAINER {
 		opts = append(opts, oci.WithLinuxNamespace(runtimespec.LinuxNamespace{Type: runtimespec.PIDNamespace, Path: GetPIDNamespace(targetPid)}))
 	}
+
+	if namespaces.GetUsernsOptions() != nil {
+		switch namespaces.GetUsernsOptions().GetMode() {
+		case runtime.NamespaceMode_NODE:
+			// Nothing to do. Not adding userns field uses the node userns.
+		case runtime.NamespaceMode_POD:
+			opts = append(opts, oci.WithLinuxNamespace(runtimespec.LinuxNamespace{Type: runtimespec.UserNamespace, Path: GetUserNamespace(sandboxPid)}))
+			opts = append(opts, oci.WithUserNamespace(uids, gids))
+		}
```

Comment:
I'd prefer every `switch{}` to have `default:` to catch potential bugs

Reply:
@AkihiroSuda do you mean to also change this function to return an error?

A switch with a default statement is added in all the rest of the cases, here I kept it like that to not change the signature and because for the PID namespace we are already ignoring "errors" (https://github.com/containerd/containerd/pull/7679/files#diff-b6195b06bf3cb9951e9681cde7e871a9585a3406255e10da96afd5fed67c0ff9R636-R638).

Or do you mean to just log something?

Reply:
maybe we should return error here

Reply:
SGTM